components/io_expander/esp_io_expander/include/esp_io_expander.h

```diff
@@ -0,0 +1,225 @@
+     *
+     * @param handle: IO Expander handle
+     * @param value: Register's value
+     *
+     * @return
+     *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+     */
+    esp_err_t (*read_direction_reg)(esp_io_expander_handle_t handle, uint32_t *value);
+
+    /**
+     * @brief Reset the device to its initial state (mandatory)
+     *
+     * @note This function will reset all device's registers
+     *
+     * @param handle: IO Expander handle
+     *
+     * @return
+     *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+     */
+    esp_err_t (*reset)(esp_io_expander_t *handle);
+
+    /**
+     * @brief Delete device (mandatory)
+     *
+     * @param handle: IO Expander handle
+     *
+     * @return
+     *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+     */
+    esp_err_t (*del)(esp_io_expander_t *handle);
+
+    /**
+     * @brief Configuration structure
+     */
+    esp_io_expander_config_t config;
+};
+
+/**
+ * @brief Set the direction of a specific IO
+ *
+ * @param handle: IO Exapnder handle
+ * @param pin_num: IO num, valid from 0 to (`io_amount` - 1)
+ * @param is_output: true - output, false - input
+ *
+ * @return
+ *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+ */
+esp_err_t esp_io_expander_set_dir(esp_io_expander_handle_t handle, uint8_t pin_num, bool is_output);
+
+/**
+ * @brief Get the direction of a specific IO
+ *
+ * @param handle: IO Exapnder handle
+ * @param pin_num: IO num, valid from 0 to (`io_amount` - 1)
+ * @param is_output: true - output, false - input
+ *
+ * @return
+ *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+ */
+esp_err_t esp_io_expander_get_dir(esp_io_expander_handle_t handle, uint8_t pin_num, bool *is_output);
+
+/**
+ * @brief Set the output level of a specific IO
+ *
+ * @note Specific IO must be in input mode first, otherwise this function will return an error
+ *
+ * @param handle: IO Exapnder handle
+ * @param pin_num: IO num, valid from 0 to (`io_amount` - 1)
+ * @param level: 0 - Low level, 1 - High level
+ *
+ * @return
+ *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+ */
+esp_err_t esp_io_expander_set_level(esp_io_expander_handle_t handle, uint8_t pin_num, uint8_t level);
+
+/**
+ * @brief Get the intput level of a specific IO
+ *
+ * @param handle: IO Exapnder handle
+ * @param pin_num: IO num, valid from 0 to (`io_amount` - 1)
+ * @param level: 0 - Low level, 1 - High level
+ *
+ * @return
+ *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+ */
+esp_err_t esp_io_expander_get_level(esp_io_expander_handle_t handle, uint8_t pin_num, uint8_t *level);
+
+/**
+ * @brief Show the current status of each IO of the device, including direction, input level and output level.
+ *
+ * @param handle: IO Exapnder handle
+ *
+ * @return
+ *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+ */
+esp_err_t esp_io_expander_show_state(esp_io_expander_handle_t handle);
```

Comment:
If show meaning print, what about rename the function to `esp_io_expander_print_state`?

Reply:
Yeah, it's better. PTAL [here](https://github.com/espressif/esp-bsp/pull/86/commits/ca919496b317741e0e9ac3e8e858dbb289b5e385#diff-946a3784b32d4d423f23b0a03eb9d05a89128cc72555733cbd528e355395c4faL199-R234).